app/javascript/gobierto_data/webapp/components/commons/SavingDialog.vue

```diff
@@ -1,120 +1,108 @@
 <template>
   <div class="gobierto-data-sql-editor-container-save">
     <!--  only show if label name is set OR the prompt is visible  -->
-    <template v-if="isSavingPromptVisible || labelValue">
-      <input
-        ref="inputText"
-        v-model="labelValue"
-        :placeholder="placeholder"
-        type="text"
-        class="gobierto-data-sql-editor-container-save-text"
-        :class="{
-          'disable-input-text': disabledButton,
-          'disable-cursor-pointer': enabledForkButton || !isUserLogged
-        }"
-        @keydown.stop="onKeyDownTextHandler"
-        @click="inputHandler"
-      >
+    <template v-if="handlerInputQuery()">
+        <input
+          ref="inputText"
+          v-model="labelValue"
+          :placeholder="placeholder"
+          type="text"
+          class="gobierto-data-sql-editor-container-save-text"
+          :class="{
+            'disable-input-text': disabledButton,
+            'disable-cursor-pointer': enabledForkButton || !isUserLogged || enabledForkVizButton
+          }"
+          @keydown.stop="onKeyDownTextHandler"
+          @click="enabledInputHandler"
+        >
     </template>
 
     <!-- only show checkbox on prompt visible -->
-    <template v-if="isForkPromptVisible">
-      <template v-if="isSavingPromptVisible">
-        <label
-          :for="labelPrivate"
-          class="gobierto-data-sql-editor-container-save-label"
+    <template v-if="enableForkPrompt()">
+      <label
+        :for="labelPrivate"
+        class="gobierto-data-sql-editor-container-save-label"
+      >
+        <input
+          :id="labelPrivate"
+          :checked="isPrivate"
+          type="checkbox"
+          @input="onInputCheckboxHandler"
         >
-          <input
-            :id="labelPrivate"
-            :checked="isPrivate"
-            type="checkbox"
-            @input="onInputCheckboxHandler"
-          >
-          {{ labelPrivate }}
-        </label>
-      </template>
+        {{ labelPrivate }}
+      </label>
+    </template>
+
 
-      <!-- only show if label name is set OR the prompt is visible -->
-      <template v-if="isSavingPromptVisible || labelValue">
-        <PrivateIcon
-          :is-closed="isPrivate"
-          :style="{ paddingRight: '.5em', margin: 0 }"
-        />
+    <!-- only show if label name is set OR the prompt is visible -->
+    <template v-if="handlerInputQuery()">
```

Comment:
Avoid function calls in the DOM, unless they're belong to events (like `@click`). That creates multiple re-renders undesiderable. It's a bad practice.

In this case:
```suggestion
    <template v-if="handlerInputQuery">
```
And move the function `handlerInputQuery` to computed block, in order to be cached.